api/src/services/import-services/sampling-periods/import-sample-periods-service.ts

```diff
@@ -0,0 +1,144 @@
+import { WorkSheet } from 'xlsx';
+import { IDBConnection } from '../../../database/db';
+import { InsertSamplePeriodObject } from '../../../repositories/sample-period-repository';
+import { CSVConfigUtils } from '../../../utils/csv-utils/csv-config-utils';
+import { validateCSVWorksheet } from '../../../utils/csv-utils/csv-config-validation';
+import { CSVConfig, CSVError } from '../../../utils/csv-utils/csv-config-validation.interface';
+import {
+  getDateCellValidator,
+  getTimeCellSetter,
+  getTimeCellValidator
+} from '../../../utils/csv-utils/csv-header-configs';
+import { getStartDateIsBeforeEndDateRowValidator } from '../../../utils/csv-utils/row-validators/start-end-date-order-row-validator';
+import { getLogger } from '../../../utils/logger';
+import { DBService } from '../../db-service';
+import { SamplePeriodService } from '../../sample-period-service';
+import { SampleSiteService } from '../../sample-site-service';
+import { TechniqueService } from '../../technique-service';
+import { getMethodTechniqueCellValidator, getSampleSiteCellValidator } from './utils/sample-periods-header-configs';
+
+const defaultLog = getLogger('services/import/import-sample-periods-service');
+
+// Sample Period CSV static headers
+export type SamplePeriodCSVStaticHeader =
+  | 'SAMPLE_SITE'
+  | 'TECHNIQUE_NAME'
+  | 'START_DATE'
+  | 'START_TIME'
+  | 'END_DATE'
+  | 'END_TIME';
+
+/**
+ * ImportSamplePeriodsService - A service for importing Sample Periods into SIMS.
+ *
+ * @class ImportSamplePeriodsService
+ * @extends DBService
+ */
+export class ImportSamplePeriodsService extends DBService {
+  worksheet: WorkSheet;
+  surveyId: number;
+
+  utils: CSVConfigUtils<SamplePeriodCSVStaticHeader>;
+
+  /**
+   * Construct an instance of ImportSamplePeriodsService.
+   *
+   * @param {IDBConnection} connection - DB connection
+   * @param {string} surveyId
+   */
+  constructor(connection: IDBConnection, worksheet: WorkSheet, surveyId: number) {
+    super(connection);
+
+    const initialConfig: CSVConfig<SamplePeriodCSVStaticHeader> = {
+      staticHeadersConfig: {
+        SAMPLE_SITE: { aliases: ['SAMPLE SITE', 'SAMPLING_SITE', 'SAMPLING SITE', 'SITE', 'LOCATION', 'STATION'] },
+        TECHNIQUE_NAME: { aliases: ['TECHNIQUE NAME', 'METHOD_TECHNIQUE', 'METHOD TECHNIQUE', 'TECHNIQUE', 'METHOD'] },
```

Comment:
nit: The name of this field is different from the observations field, and the array of aliases is also different.

I'm sure it doesn't change the functionality, but might be good to unify them if we can.

Reply:
Same with 'SAMPLING_SITE' vs 'SAMPLING_SITE', etc.

(Possibly you mentioned doing that in another PR later, if so, nvm!)

Reply:
Updated.